restli-client/src/main/java/com/linkedin/restli/client/base/EntityActionRequestBuilderBase.java

```diff
@@ -0,0 +1,27 @@
+package com.linkedin.restli.client.base;
+
+import com.linkedin.restli.client.ActionRequest;
+import com.linkedin.restli.client.RestliRequestOptions;
+import com.linkedin.restli.common.ResourceSpec;
+
+/**
+ * The abstract base class for all generated request builders classes for entity-level actions.
+ */
+public abstract class EntityActionRequestBuilderBase<K, V, RB extends ActionRequestBuilderBase<K, V, RB>>
+        extends ActionRequestBuilderBase<K, V, RB>
+{
+    protected EntityActionRequestBuilderBase(String baseUriTemplate, Class<V> elementClass, ResourceSpec resourceSpec,
+        RestliRequestOptions requestOptions)
+    {
+      super(baseUriTemplate, elementClass, resourceSpec, requestOptions);
+    }
+
+    @Override
+    public ActionRequest<V> build() {
+      if (!isIdPresent())
+      {
+        throw new IllegalStateException("Missing or null id value; we cannot invoke this entity-level action.");
```

Comment:
nit: I suggest re-wording this message to say something like `Entity-level action request is missing required id value.`

Reply:
Done, thanks.